Objects/stringlib/unicode_format.h

```diff
@@ -462,8 +462,21 @@ get_field_object(SubString *input, PyObject *args, PyObject *kwargs,
             tmp = getattr(obj, &name);
         else
             /* getitem lookup "[]" */
-            if (index == -1)
+            if (index == -1) {
+                if (PySequence_Check(obj)) {
+                    /* string index can't be passed to sequence */
+                    PyObject *str = SubString_new_object(&name);
+                    if (str) {
+                        PyErr_Format(PyExc_TypeError,
+                             "Index must be a sequence of digits, string %R "
+                             "is not allowed",
+                             str);
+                    }
+                    Py_XDECREF(str);
+                    goto error;
```

Comment:
It seems that whether the above if is true or not, it will jump to an error. And to trigger an exception, you need to return `NULL`.
Maybe it's because I can't show the complete context here, which makes me a little misunderstood.